hackforla/VRMS
# Add testing for application (backend, frontend, end2end)

Adds in frontend testing for the application. This branch was branched from  https://github.com/hackforla/VRMS/pull/264. 

Overview:
Our application is in some dire need of some repeatability and testing is just the thing to remedy this. The goal of this branch started with getting our unit tests working again. It then moved into a discussion of what would help the team, which came up as adding something like Cypress. In all, this branch makes things much more repeatable and tested! 

Acceptance Criteria:
[x] We have working Jest tests in the frontend
[x] We have working Jest tests in the backend
[x] We have a working end-2-end test suite (likely Cypress)
[x] Make sure that all test runners are working as expected.
[x] Configurable environment variables
[x] Update the README and relevant end documentation.

Setup & Verification Steps:
1. Pull my branch locally
 - git remote add nickb https://github.com/nickbeaird/VRMS.git
 - git checkout -b test_nicks_tests 
 - git fetch nickbeaird
 - git branch --set-upstream test_nicks_tests nickb/add_front_end_testing
 - git pull

* Please verify that the changes have been made by verifying that you have a "backend" and "cypress" directory 

2. Move your .env file at the root directory to the backend directory

3. Add the below variables to your backend/.env file
BACKEND_PORT=YOUR_DESIRED_PORT
REACT_APP_PROXY=http://localhost:${BACKEND_PORT}
SLACK_OAUTH_TOKEN=ASK_NICK_FOR_HIS?
SLACK_BOT_TOKEN=ASK_NICK_FOR_HIS?
SLACK_TEAM_ID=ASK_NICK_FOR_HIS?
SLACK_CHANNEL_ID=ASK_NICK_FOR_HIS?
SLACK_CLIENT_ID=ASK_NICK_FOR_HIS?
SLACK_CLIENT_SECRET=ASK_NICK_FOR_HIS?
SLACK_SIGNING_SECRET=ASK_NICK_FOR_HIS?

4. Add the below variables to your client/.env file
CLIENT_PORT=YOUR_DESIRED_PORT
CLIENT_URL=http://localhost:${CLIENT_PORT}

5. Delete all of your package.json file and nod_modules directories

6. Run yarn install in the root, backend, and client directories. 
 - cd root && yarn install 
 - cd client && yarn install 
 - cd ../backend && yarn install

7. Navigate to the root directory
 - cd ..

8. Run all the tests with **npm run test:all**

The above should have a fully functional running sets of test suites
